app/controllers/admin/web_stats_controller.rb

```diff
@@ -9,20 +9,23 @@
 # modify it under the terms of the GPL (version 2 or later).
 # ============================================================================
 class Admin::WebStatsController < AdminController
-  def index
-    @visits = filtered_visits
-    authorise Ahoy
-    authorise @visits if @visits.present?
-  end
+  before_action :set_ahoy_user
 
-  def filtered_visits
+  def index
     page_num = params[ :page ] || 1
     visits = Ahoy::Visit
-    if params[:user_id].present?
-      @user = User.find( params[:user_id] )
-      visits = visits.where( user: @user )
-    end
+    visits = visits.where( user: @ahoy_user ) if @ahoy_user
     @visits = visits.order( 'started_at desc' ).page( page_num )
-    authorise @visits
+
+    authorise Ahoy::Visit
```

Comment:
As above, consider throwing out unauthorised users asap by moving `authorise Ahoy::Visit` to the first line of the action.